### PR TITLE
fs: littlefs: add littlefs disk version selection

### PR DESCRIPTION
**Pull request by MikhailSiomin**
Add the ability to select littlefs disk version to maintain backward compatibility with existing littlefs with the same major disk version.

### CODE REVIEW

**Review by de-nordic — CHANGES_REQUESTED** [review]
Change in LittleFS got NACK, there is no need to make change there.
The macros are only needed here.
Zephyr module of littlefs is synchronized from upstream LittleFS and changes done in our module are not required for it to work with Zephyr.

I do not mind change in the LittleFS Zephyr support code, this seems ok.

File: subsys/fs/littlefs_fs.c [hunk 1]
```diff
@@ -882,6 +890,12 @@ static int littlefs_init_cfg(struct fs_littlefs *fs, int flags)
 		lcp->sync = lfs_api_sync;
 	}
 
+#ifdef CONFIG_FS_LITTLEFS_DISK_VERSION
+	lcp->disk_version = disk_version;
+	LOG_INF("partition disk version: %u.%u", LFS_DISK_VERSION_MAJOR,
```
Comment (MikhailSiomin) [inline subsys/fs/littlefs_fs.c:895]: I think it's better to change `LFS_DISK_VERSION_MAJOR` by `FS_LITTLEFS_DISK_VERSION_MAJOR_GET(disk_version)`.

Comment (de-nordic) [reply subsys/fs/littlefs_fs.c:895]: I also think so, you may also report here what is actual driver supported version.

Comment (MikhailSiomin) [reply subsys/fs/littlefs_fs.c:895]: @de-nordic, the `littlefs_init_fs` function reports the actual supported disk version. Will this be enough?

Comment (de-nordic) [reply subsys/fs/littlefs_fs.c:895]: Yes.

**Review by de-nordic — DISMISSED** [review]

**Comment by MikhailSiomin** [comment]
@decsny, @galak, @Laczen, @nashif, can you please review this PR?

**Review by pdgendt — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by rruuaanng — APPROVED** [review]

**Pull request merged**